apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -87,19 +92,31 @@ const InnerFileBrowser = React.memo(
                       moveFilePrompt={moveFilePrompt}
                       renameFilePrompt={renameFilePrompt}
                       renameFolderPrompt={renameFolderPrompt}
+                      toggleFileVisibility={toggleFileVisibility}
                     />
                   </ul>
                 )}
               </li>
             );
           })}
         {Object.values(files)
-          .filter(f => f.folderId === parentId)
+          .filter(
+            f => f.folderId === parentId && (f.visible !== false || isStartMode)
+          )
           .sort((a, b) => a.name.localeCompare(b.name))
           .map(f => (
             <li key={f.id}>
               <span className="label">
-                <span onClick={() => openFile(f.id)}>{f.name}</span>
+                <span onClick={() => openFile(f.id)}>
+                  {isStartMode && (
+                    <i
+                      className={`fa-solid ${
```

Comment:
would it be worth making a helper method for getting the icon based on the visibility? I see that repeated a few places.

Reply:
I'm not sure. The logic is repeated in two places with the opposite result for each. E.g. if the file is hidden, label shows a closed eye but the button bar has an open eye. We could have a boolean parameter to specify which we want, but I wonder if it's worth it. I think it's also likely that we're going to cover the button bar to a single ellipsis icon with a dropdown menu, which should remove the similar logic from one of two places.

Reply:
that's fair, makes sense to leave as is

Reply:
FWIW, switching to `.hidden` makes it slightly less complicated too.